distsql/request_builder_test.go

```diff
@@ -529,6 +529,7 @@ func TestRequestBuilder7(t *testing.T) {
 		{kv.ReplicaReadFollower, "Follower"},
 		{kv.ReplicaReadMixed, "Mixed"},
 	} {
+		replicaRead := replicaRead
```

Comment:
Please create a variable with a different name, to make it clear it is a copy, instead of overloading the variable name. And/or add a comment on why this is needed, so it is less confusing for developers unknown of the issue.

Reply:
this paradigm is so common that i believe use the same name is the best/acceptable practise
see 
- https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks
- https://github.com/golang/go/wiki/CommonMistakes#introduction

maybe add comment to make it more clear

Reply:
added